### PR TITLE
Improve grammar and wording

### PR DESCRIPTION
**Pull request by mosabua**
## Description

To enable better understanding of how to write routing rules.

Update from @andythsu s PR for pending status .. once I saw the file I just updated it all instead of just a small section.

Note for reviewers - this is all just grammar and wrapping and similar changes. Nothing fundamentally changed in terms of content and technical information.

## Additional context and related issues

#360 

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/routing-rules.md [hunk 1]
```diff
@@ -132,73 +138,77 @@ You can use the `contains` operator
...
+The  `TrinoRequestUser` class attempts to extract user information from a
+request, in the following order:
 
-1. The `X-Trino-User` header
-2. The `Authorization: Basic` header
-3. The `Authorization: Bearer` header. Requires configuring an OAuth2 User Info URL
-4. The `Trino-UI-Token` or `__Secure-Trino-ID-Token` cookie
+1. `X-Trino-User` header.
+2. `Authorization: Basic` header.
+3. `Authorization: Bearer` header. Requires configuring an OAuth2 User Info URL
```
Comment (willmostly) [inline docs/routing-rules.md:167]:
```suggestion
3. `Authorization: Bearer` header. Requires configuring an OAuth2 User Info URL.
```

File: docs/routing-rules.md [hunk 2]
```diff
@@ -132,73 +138,77 @@ You can use the `contains` operator
...
-view will not be expanded, and tables referenced by the view will not be
-recognized. Note that Views and Materialized Views are treated as tables and
-added to the list of tables in all contexts, including  statements such as
-`CREATE VIEW`.
+The `TrinoQueryProperties` class attempts to parse the body of a request to
+determine the SQL statement and other information. Note that only a
+syntactic analysis is performed.
+
+If a query references a view, then that view  is expanded, and tables referenced
+by the view are recognized. Views and materialized views are treated as
```
Comment (willmostly) [inline docs/routing-rules.md:201]:
```suggestion
If a query references a view, then that view is not expanded, and tables referenced
by the view are not recognized. Views and materialized views are treated as
```
This is what is implied by syntactic-only analysis. The Trino Gateway does not reach out to the backend to get information on view and table definitions

File: docs/routing-rules.md [hunk 3]
```diff
@@ -227,50 +237,52 @@ object:
...
 When extracting the user from a JWT token, this field is used as the username.
 By default, the `email` claim is used. 
  
-#### oauthTokenInfoUrl
+`oauthTokenInfoUrl`:
 
-If configured, then Trino will attempt to retrieve user info by exchanging
-potential authorization tokens with this URL. Responses are cached for 10
-minutes to avoid triggering rate limits.
+If configured, Trino attempts to retrieve the user info by exchanging potential
```
Comment (willmostly) [inline docs/routing-rules.md:270]:
```suggestion
If configured, Trino Gateway attempts to retrieve the user info by exchanging potential
```

File: docs/routing-rules.md [hunk 4]
```diff
@@ -227,50 +237,52 @@ object:
...
+All rules whose conditions are satisfied fire. For example, in the "airflow"
+and "airflow special" example rules from the following rule priority section, a
+query with source `airflow` and label `special` satisfies both rules. The
+`routingGroup` is set to `etl` and then to `etl-special` because of the order in
+which the rules of defined. If you swap the order of the rules, then you get
+`etl` instead.
 
-One could solve this by writing the rules such that they're atomic (any query
-will match exactly one rule). For example we can change the first rule to
+You can avoid this ordering issue by writing atomic the rules, so any query
```
Comment (willmostly) [inline docs/routing-rules.md:283]:
```suggestion
You can avoid this ordering issue by writing atomic rules, so any query
```

File: docs/routing-rules.md [hunk 5]
```diff
@@ -112,15 +118,15 @@ object called `request`. Rules may also utilize
 [trinoRequestUser](#trinorequestuser) and
 [trinoQueryProperties](#trinoqueryproperties)
 objects, which provide information about the user and query respectively.
-There should be at least one action of the form
-`result.put(\"routingGroup\", \"foo\")` which says that if a request satisfies
-the condition, it should be routed to `foo`.
+There must be at least one action of the form
+`result.put(\"routingGroup\", \"foo\")` to configure routing to the `foo`
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/routing-rules.md]: Updated.. I think the new wording is clearer

**Review by willmostly — APPROVED** [review]
LGTM!

**Pull request merged**